packages/animations/lib/src/page_transition_switcher.dart

```diff
@@ -246,23 +336,25 @@ class _PageTransitionSwitcherState extends State<PageTransitionSwitcher>
       // update the transition instead of replacing it.
       _currentEntry.widgetChild = widget.child;
       _updateTransitionForEntry(_currentEntry); // uses entry.widgetChild
+      _markChildWidgetCacheAsDirty();
     }
   }
 
   void _addEntryForNewChild({@required bool shouldAnimate}) {
     assert(shouldAnimate || _currentEntry == null);
     if (_currentEntry != null) {
       assert(shouldAnimate);
+      assert(!_outgoingEntries.contains(_currentEntry));
       if (widget.reverse) {
         _currentEntry.primaryController.reverse();
       } else {
         _currentEntry.secondaryController.forward();
       }
+      _outgoingEntries.add(_currentEntry);
```

Comment:
This will not work: It will cause the order of widgets to change unexpectedly if _currentEntry was previously shown below all `_outgoingEntries` instead of on top of it.

Reply:
Maybe the layout builder for this widget should just get a List of widgets without knowledge of the current entry, to avoid this problem?

Reply:
So the old activeEntries list, right?